build.rs

```diff
@@ -1,11 +1,55 @@
 extern crate bindgen;
 
+use std::env;
+use std::io;
 use std::path::PathBuf;
 use std::process::Command;
 
 const HEADER_PATH: &str = "include/uapi/linux/nitro_enclaves.h";
 const OUT_FILE: &str = "/driver_structs.rs";
 
+// Generate the sysroot path based on the architecture the program is built.
+// This is needed for the compiler to locate the architecture dependent
+// header files.
+fn get_gcc_sysroot() -> String {
+    // Initialize the output because it might remain uninitialized with the
+    // current control flow
+    let mut output = Err(io::Error::from(io::ErrorKind::NotFound));
+    let mut cc_found = false;
+
+    // Use the `CC` environment variable if existing
+    // which should give us directly the compiler's name
+    let cc = env::var("CC");
+    if let Ok(cc) = cc {
+        output = Command::new(cc).arg("--print-sysroot").output();
+        if let Ok(_) = output {
+            cc_found = true;
+        }
+    }
+
+    if !cc_found {
```

Comment:
why isn't this on the else branch ?

Reply:
Because there are two situations that can fail:
1. Environment variable "CC" is not set
2. The command "$CC --print-sysroot" fails for any reason (such as $CC is not set correctly)

If I just add the code on the else statement of the first if, the second case would not be covered.

Reply:
Rigght but it could be:
if let Ok(cc) = cc {
....
}
if let Err(_) = output {
....
}
as you did bellow

Reply:
Yep, here I agree :D Will change it.